cmd/embedded-cluster/util.go

```diff
@@ -0,0 +1,65 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/replicatedhq/embedded-cluster/pkg/helpers"
+)
+
+// createSystemdUnitFiles links the k0s systemd unit file. this also creates a new
+// systemd unit file for the local artifact mirror service.
+func createSystemdUnitFiles(isWorker bool, proxy *Proxy) error {
+	dst := systemdUnitFileName()
+	if _, err := os.Lstat(dst); err == nil {
+		if err := os.Remove(dst); err != nil {
+			return err
+		}
+	}
+	src := "/etc/systemd/system/k0scontroller.service"
+	if proxy != nil {
+		ensureProxyConfig(fmt.Sprintf("%s.d", src), proxy.HTTPProxy, proxy.HTTPSProxy, proxy.NoProxy)
+	}
+	if isWorker {
+		src = "/etc/systemd/system/k0sworker.service"
+		if proxy != nil {
+			ensureProxyConfig(fmt.Sprintf("%s.d", src), proxy.HTTPProxy, proxy.HTTPSProxy, proxy.NoProxy)
+		}
+	}
```

Comment:
```suggestion
	src := "/etc/systemd/system/k0scontroller.service"
	if isWorker {
		src = "/etc/systemd/system/k0sworker.service"
	}
	if proxy != nil {
		ensureProxyConfig(fmt.Sprintf("%s.d", src), proxy.HTTPProxy, proxy.HTTPSProxy, proxy.NoProxy)
	}
```

Unless you were intending to create two proxy config files on worker nodes?

Reply:
I've seen that `/etc/systemd/system/k0sworker.service` isn't doing anything, but it might be that this has been fixed while working on the proxy?